tests/scripts/all.sh

```diff
@@ -1044,7 +1044,10 @@ component_check_test_dependencies () {
         tests/suites/test_suite_psa*.data tests/suites/test_suite_psa*.function |
         grep -Eo '!?MBEDTLS_[^: ]*' |
         grep -v MBEDTLS_PSA_ |
-        sort -u > $found
+        # By default, sort (v8.25) on ubuntu-16 and sort (v8.30) on ubuntu-20
+        # sort text in different order. We use -d option to sort text in
+        # an order considering only blanks and alphanumeric characters.
+        sort -ud > $found
```

Comment:
`sort -ud` would consider `FOO` and `!FOO` the same and omit the one with later occurrence.

I'm not sure if it is major, since we are seemly to have very low possibility to have both `FOO` and `!FOO` dependencies in psa tests.

In fact, the different results are related to [reference](https://stackoverflow.com/questions/76331773/how-to-get-the-sort-shell-command-to-compare-raw-bytes):
> Collation order and (multi-byte) character type are influenced by your locale.

So, I would suggest to use `sort -u | sort -d > $found` or `LC_ALL=C sort -u > $found`.

Wdyt?

Reply:
Good point! I think I like `LC_ALL_C sort -u` better if it works, otherwise `LC_ALL=C sort -u | sort -d`.

Reply:
`-d` doesn't protect against locale differences. Please use `LC_ALL=C sort …` or add `export LC_COLLATE=C` to `pre_initialize_variables`. (Not `LC_ALL=C` because I think we should preserve at least `LC_CTYPE`.)